compiler/env/OMRClassEnv.cpp

```diff
@@ -51,7 +51,13 @@ OMR::ClassEnv::getArrayElementWidthInBytes(TR::Compilation *comp, TR_OpaqueClass
 intptr_t
 OMR::ClassEnv::getVFTEntry(TR::Compilation *comp, TR_OpaqueClassBlock* clazz, int32_t offset)
    {
-   return *(intptr_t*) (((uint8_t *)clazz) + offset);
+   // There is no project-agnostic way to determine whether or not offset is a
+   // valid VFT offset for clazz, so return 0 to be safe. If offset were valid,
+   // the result would be:
+   //
+   //    *(intptr_t*) (((uint8_t *)clazz) + offset)
+   //
+   return 0;
```

Comment:
`TR_UNIMPLEMENTED()` is a call to `TR::fatal_assertion()`, which will crash, forcing any downstream project to override this method. To me that seems a bit drastic in cases like this one, where an override might be desirable but is not strictly necessary

Reply:
I guess I'm having difficulty imagining why a downstream project would want to use that default implementation, but they'd likely run into plenty of other problems pretty early on if this implementation didn't suffice, so adding an unconditional assertion probably won't save them any debugging effort.

Reply:
> they'd likely run into plenty of other problems pretty early on if this implementation didn't suffice

What kind of problems do you expect? At least for the moment (and AFAICT ever since its introduction years ago), `getVFTEntry()` is used only to try to optimize guards with method tests in the VP handler that I'm modifying in this PR. With the other changes in this PR, getting back a zero will just inhibit this very particular optimization.

Reply:
I was imagining a downstream project might want to use it for some other reason besides this optimization, but if it always returned zero, it would be pretty useless for them, and they'd be forced to implement it.  But as you point out, the default implementation is safe in the context of this optimization, and it wouldn't force the downstream project to override it all.

Thanks for answering the question!